kernel/arch/aarch64/boot/exc_dispatch.c

```
@@ -0,0 +1,35 @@
+
+void aarch64_exc_register_handler(int slot, void (*handler)(void)) {
+    /* Registration is a boot-time operation: callers must install handlers
+     * before unmasking IRQs with DAIFCLR so dispatch can read lock-free. */
+    if (slot >= 0 && (uint32_t)slot < MAX_HANDLERS)
+        s_handlers[slot] = handler;
+}
+
+/* Called from each vector slot with the slot index in x0 (as uint64_t
+ * per the AArch64 calling convention; narrowed to int safely since 0-15). */
+void aarch64_exc_dispatch(uint64_t slot) {
+    int n = (int)(slot & 0xFu);
```

Clang (14.0.6)

[warning] variable name 'n' is too short, expected at least 3 characters

(readability-identifier-length)